src/App.tsx

```diff
@@ -1,24 +1,172 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import {
+  addTodos,
+  deleteTodos,
+  editTodos,
+  getTodos,
+} from './api/todos';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Notification } from './components/Notification';
+import { TodoItem } from './components/TodoItem';
+import { Errors } from './types/Errors';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+const USER_ID = 6398;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedStatus, setSelectedStatus] = useState(TodoStatus.All);
+  const [errorType, setErrorType] = useState(Errors.None);
+  const [hasErrorNotification, setHasErrorNotification] = useState(false);
+  const [isUpdatingIds, setIsUpdatingIds] = useState<number[]>([]);
+
+  const hideErrorNotifications = () => setHasErrorNotification(false);
+  const showErrorNotification = (error:Errors) => {
+    setErrorType(error);
+    setHasErrorNotification(true);
+    setTimeout(hideErrorNotifications, 3000);
+  };
+
+  const fetchAllTodos = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      showErrorNotification(Errors.Get);
+    }
+  };
+
+  useEffect(() => {
+    fetchAllTodos();
+  }, []);
+
+  const addNewTodo = async (title: string) => {
+    try {
+      const newTodo = {
+        title,
+        userId: USER_ID,
+        completed: false,
+      };
+
+      await addTodos(newTodo);
+      await fetchAllTodos();
+    } catch {
+      showErrorNotification(Errors.Add);
+    }
+  };
+
+  const deleteTodoFromServer = async (todoId:number) => {
+    try {
+      await deleteTodos(todoId);
+      await fetchAllTodos();
+    } catch {
+      showErrorNotification(Errors.Delete);
+    }
+  };
+
+  const updateTodo = async (
+    todo: Todo,
+    key:keyof Todo,
+    value: string | boolean,
+  ) => {
+    const editedTodo = {
+      ...todo,
+      [key]: value,
+    };
+
+    setIsUpdatingIds((currentIds) => [...currentIds, todo.id]);
+
+    try {
+      await editTodos(todo.id, editedTodo);
+      await fetchAllTodos();
+      // setIsLoaded(true);
+      setIsUpdatingIds([]);
+    } catch (error) {
+      showErrorNotification(Errors.Edit);
+    }
+  };
+
+  let visibleTodos = [...todos];
+
+  if (selectedStatus === TodoStatus.Active) {
```

Comment:
if you need to process some logic according to the value of some variable, will be better to use switch / case
and let's move the code for preparing todos in separate function